### PR TITLE
Add host functions for CAP-52

### PR DESCRIPTION
**Pull request by jonjove**
### What

Removes stale `get_last_operation_result` host function that no longer exists. Adds host functions specified in CAP-52 without implementations.

### Why

The CAP-52 host functions are need for the implementation of CAP-54.

### Known limitations

New host functions aren't actually implemented yet.

### CODE REVIEW

File: stellar-contract-env-common/src/env.rs [hunk 1]
```diff
@@ -183,6 +183,13 @@ macro_rules! call_macro_with_all_host_functions {
                 {"_", fn compute_hash_sha256(x:Object) -> Object}
                 {"0", fn verify_sig_ed25519(x:Object, k:Object, s:Object) -> RawVal}
             }
+
+            mod account "a" {
+                {"_", fn get_low_threshold(a:Object) -> RawVal}
```
Comment (leighmcculloch) [inline stellar-contract-env-common/src/env.rs:188]: Could you add an `account_` prefix to these? We're using an object name prefix on host fns for most object types.

Comment (jonjove) [reply stellar-contract-env-common/src/env.rs]: Done.

**Review by leighmcculloch — COMMENTED** [review]
I forgot to ask, can you add comments to the new functions so we don't need to double back and add them later? The macro contents support doc comments. Have a look at https://github.com/stellar/rs-stellar-contract-env/pull/131/files#diff-2706694cc99e5d5001390a8e291d5ea22144fd9a7b1c6f872c4cfaec61bbc0b9R87 which adds comments for some existing types.

**Comment by jonjove** [comment]
@leighmcculloch done.

File: stellar-contract-env-common/src/env.rs [hunk 2]
```diff
@@ -183,6 +186,24 @@ macro_rules! call_macro_with_all_host_functions {
                 {"_", fn compute_hash_sha256(x:Object) -> Object}
                 {"0", fn verify_sig_ed25519(x:Object, k:Object, s:Object) -> RawVal}
             }
+
+            mod account "a" {
+                /// Get the low threshold for the account with ed25519 public
+                /// key a (a is binary). Traps if no such account exists.
```
Comment (leighmcculloch) [inline stellar-contract-env-common/src/env.rs:192]: >Traps if no such account exists.

Any trap needs to be avoidable, so I guess we need a function such as `account_does_exist`.

**Review by leighmcculloch — APPROVED** [review]

File: stellar-contract-env-common/src/env.rs (hunk 2 above)
Comment (jonjove) [reply stellar-contract-env-common/src/env.rs:192]: Agreed. I realized we need the same for `has_invoking_contract`. I'll get to that when I actually start consuming these functions next week.

**Pull request merged**